src/main/java/tempest/ui/GUIManager.java

```diff
@@ -64,15 +76,20 @@ public void windowClosing(WindowEvent e) {
      *
      * @param name Page's name
      */
-    public void checkPage(String name) {
-        System.out.println(name);
-
-        if (name.equals(PageNames.MANAGE_MODULES)) {
-            ManageModulesPage mmp = (ManageModulesPage) getPage(ManageModulesPage.class);
-            mmp.update();
+    private void checkPage(String name) {
```

Comment:
This would be better named as updatePage rather than checkPage, as this method is itself doing the updating.